nexus-repository-r/src/test/java/org/sonatype/nexus/repository/r/internal/RPackagesInformationBuilderFacetImplTest.java

```diff
@@ -0,0 +1,285 @@
+    when(assetCreatedEvent.getRepositoryName()).thenReturn(REPOSITORY_NAME);
+    when(assetCreatedEvent.getAsset()).thenReturn(asset);
+    when(asset.name()).thenReturn(PACKAGES_GZ_PATH);
+    when(formatAttributes.get(P_ASSET_KIND, String.class)).thenReturn(PACKAGES.name());
+
+    underTest.on(assetCreatedEvent);
+
+    verifyNoMoreInteractions(eventManager);
+  }
+
+  @Test
+  public void testAssetUpdatedEventHandledCorrectly() {
+    when(assetUpdatedEvent.isLocal()).thenReturn(true);
+    when(assetUpdatedEvent.getRepositoryName()).thenReturn(REPOSITORY_NAME);
+    when(assetUpdatedEvent.getAsset()).thenReturn(asset);
+    when(asset.name()).thenReturn(ASSET_PATH);
+    when(formatAttributes.get(P_ASSET_KIND, String.class)).thenReturn(ARCHIVE.name());
+
+    underTest.on(assetUpdatedEvent);
+
+    ArgumentCaptor<RMetadataInvalidationEvent> eventCaptor = ArgumentCaptor.forClass(RMetadataInvalidationEvent.class);
+    verify(eventManager).post(eventCaptor.capture());
+
+    RMetadataInvalidationEvent event = eventCaptor.getValue();
+    assertThat(event.getRepositoryName(), is(REPOSITORY_NAME));
+    assertThat(event.getBasePath(), is(BASE_PATH));
+  }
+
+  @Test
+  public void testAssetUpdatedEventIgnoredForDifferentNode() {
+    when(assetUpdatedEvent.isLocal()).thenReturn(false);
+    when(assetUpdatedEvent.getRepositoryName()).thenReturn(REPOSITORY_NAME);
+    when(assetUpdatedEvent.getAsset()).thenReturn(asset);
+    when(asset.name()).thenReturn(ASSET_PATH);
+    when(formatAttributes.get(P_ASSET_KIND, String.class)).thenReturn(ARCHIVE.name());
+
+    underTest.on(assetUpdatedEvent);
+
+    verifyNoMoreInteractions(eventManager);
+  }
+
+  @Test
+  public void testAssetUpdatedEventIgnoredForDifferentRepository() {
+    when(assetUpdatedEvent.isLocal()).thenReturn(true);
+    when(assetUpdatedEvent.getRepositoryName()).thenReturn("foo");
+    when(assetUpdatedEvent.getAsset()).thenReturn(asset);
+    when(asset.name()).thenReturn(ASSET_PATH);
+    when(formatAttributes.get(P_ASSET_KIND, String.class)).thenReturn(ARCHIVE.name());
+
+    underTest.on(assetUpdatedEvent);
+
+    verifyNoMoreInteractions(eventManager);
+  }
+
+  @Test
+  public void testAssetUpdatedEventIgnoredForPackagesGzAsset() {
+    when(assetUpdatedEvent.isLocal()).thenReturn(true);
+    when(assetUpdatedEvent.getRepositoryName()).thenReturn(REPOSITORY_NAME);
+    when(assetUpdatedEvent.getAsset()).thenReturn(asset);
+    when(asset.name()).thenReturn(PACKAGES_GZ_PATH);
+    when(formatAttributes.get(P_ASSET_KIND, String.class)).thenReturn(PACKAGES.name());
+
+    underTest.on(assetUpdatedEvent);
+
+    verifyNoMoreInteractions(eventManager);
+  }
+
+  @Test
+  public void testRebuildMetadataOnEvent() throws Exception {
+    when(invalidationEvent.getBasePath()).thenReturn(BASE_PATH);
+    when(invalidationEvent.getRepositoryName()).thenReturn(REPOSITORY_NAME);
+
+    underTest.on(invalidationEvent);
+
+    verify(hostedFacet).buildAndPutPackagesGz(BASE_PATH);
+  }
+}
```

Comment:
New line after

Reply:
done in fe40919080a00ea9162d0af9be02c414af338e27